### ISSUE TITLE
Detect if sufficient storage space is available prior to exporting files

### ISSUE THREAD

**Issue by eloquence** [freedomofpress/securedrop-workstation#292]
It's not fun to prepare a large file export, only to find out that it's not running to completion due to insufficient disk space. It's especially problematic in our Qubes workflow, where users most can't easily fix things on their own.

Ideally we should find a way to detect whether sufficient space is available on the target device before running the export. This may be at a later stage than the pre-flight check (#281) and should be considered a stretch goal ("should") for the pilot roll-out.

**Comment by eloquence** [comment]
One wild idea: Instead of using `cp -r`, why not invoke Nautilus, which will a) perform this check for us prior to a copy operation, b) handle other issues (like filename conflicts, #270) along with it?

This seems to be possible at least via `dbus` on Ubuntu 18.04:

```bash
$ dbus org.gnome.Nautilus /org/gnome/Nautilus org.gnome.Nautilus.FileOperations.CopyFile "file:////some/dir/some-file" "some-file" "file:///some/dir/some-file" "some-file" 
```

Note that at least on Ubuntu, this does _not_ bring up a whole file manager (nor provide obivous access to one), but it will show interactive warning/error messages if there are problems during the operation.

**Comment by ninavizz** [comment]
Would the above expose users to Nautilus? If not, would a filename conflict need to be returned to the user in a Client error?

**Comment by eloquence** [comment]
As far as I can tell, it would only expose them to specific dialogs for resolving problems with the copy operation, e.g.:

![Screenshot from 2019-07-17 16-47-05](https://user-images.githubusercontent.com/213636/61419303-875a5500-a8b2-11e9-9f81-8d8a2b113917.png)

**Comment by ninavizz** [comment]
a) Would that dialog appear in the Client or the External VM?
b) That's some sh**y dialog text; it's a "disk" not a "destination," and it's the mental model for "Save" more than "Copy" I'd bet.

I'd much prefer keeping all error dialogs w/in our overlays in the Client. My goal with the flowchart is to keep things as direct, contextual, and conversational as possible.

**Comment by eloquence** [comment]
> a) Would that dialog appear in the Client or the External VM?

In the external VM. I agree we'd give up some flexibility on the presentation, but we might want to accept that trade-off if that gives us the functionality for free -- including, e.g., human-friendly presentation of detailed information, "Retry" option, etc.

**Comment by ninavizz** [comment]
The "In the External VM" part of it worries me the most, much more than the aesthetic points that have me grabbing for my `designSnob` smelling salts, tbh. Could something at least get passed back to the Client VM, or would it auto-layer the External VM atop the Client VM?

I absolutely agree wrt prioritzation for Beta. We want user feedback on as much as possible... I just don't want to leave users hanging, staring at the Client VM and wondering what's happening when nothing is going on in that frame.

**Comment by ninavizz** [comment]
(also: the above window I would _not_ qualify as human-friendly... and I suspect will be really confusing. Not in a showstopper kind of a way I suspect, but in a `wtf` kind of a way that too many are just too comfortable with, in FOSS. Not a dragon to slay for the Pilot, but something I do want folks to be mindful of.)

**Comment by eloquence** [comment]
I'd like to get a basic sanity check on this idea from @emkll before rabbit-holing too much on the UX :)

**Comment by ninavizz** [comment]
Of course. Sorry, most of my afternoon has been a deep, dark rabbithole into printing and export UX, and this just kinda surprised me. :)
